File: primer4/Product.js
class Product {
    #id;
    #name;
    #price;
    #quantity;
  
    constructor(id, name, price, quantity) {
      this.#id = id;
      this.#name = name;
      this.#price = price;
      this.#quantity = quantity;
    }
  
    get id() {
      return this.#id;
    }
    get name() {
      return this.#name;
    }
    get price() {
      return this.#price;
    }
  
    get quantity() {
      return this.#quantity;
    }
    set quantity(newQuantity) {
      this.#quantity = newQuantity;
    }
  
    getProductDetails() {
      return {
        id: this.#id,
        name: this.#name,
        price: this.#price,
        quantity: this.#quantity,
      };
    }
  }

  export default Product;

File: primer4/Product.py
class Product:
    def __init__(self, id, name, price, quantity):
        self._id = id
        self._name = name
        self._price = price
        self._quantity = quantity
    
    @property
    def id(self):
        return self._id
    
    @property
    def name(self):
        return self._name
    
    @property
    def price(self):
        return self._price
    
    @property
    def quantity(self):
        return self._quantity
    
    @quantity.setter
    def quantity(self, new_quantity):
        self._quantity = new_quantity
    
    def get_product_details(self):
        return {
            "id": self._id,
            "name": self._name,
            "price": self._price,
            "quantity": self._quantity
        }